arch/x86/crc_folding.c

```diff
@@ -249,7 +249,7 @@ ZLIB_INTERNAL void crc_fold_copy(deflate_state *const s, unsigned char *dst, con
         goto partial;
     }
 
-    algn_diff = ((uintptr_t)0 - (uintptr_t)src) & 0xF;
+    algn_diff = (uintptr_t)src % 16;//((intptr_t)0 - (intptr_t)src) & 0xF;
```

Comment:
I don't see why 16-n mod 15 would give same result as n mod 15...

Reply:
The original line hit a UBSAN error about unsigned overflow. It must be a remnant of my debugging.

Reply:
Fixing it should be separate pull request. That way it can be corrected without affecting the result. Basically the last bits need to be inverted. That is what 0-n basically does.

Reply:
I have reverted the change. I don't understand why it would need to change MSB if it was going to do `& 0xF`.